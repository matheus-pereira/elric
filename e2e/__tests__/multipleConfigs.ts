/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {wrap} from 'elric-snapshot-serializer-raw';
import slash = require('slash');
import {extractSummary} from '../Utils';
import runelric from '../runelric';

const MULTIPLE_CONFIGS_WARNING_TEXT = 'Multiple configurations found';

test('multiple configs will warn', () => {
  const rootDir = slash(path.resolve(__dirname, '../..'));
  const {exitCode, stderr} = runelric('multiple-configs', [], {
    skipPkgJsonCheck: true,
  });

  expect(exitCode).toBe(0);
  expect(stderr).toContain(MULTIPLE_CONFIGS_WARNING_TEXT);

  const cleanStdErr = stderr.replace(new RegExp(rootDir, 'g'), '<rootDir>');
  const {rest, summary} = extractSummary(cleanStdErr);

  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('multiple configs warning can be suppressed by using --config', () => {
  const {exitCode, stderr} = runelric(
    'multiple-configs',
    ['--config', 'elric.config.json'],
    {
      skipPkgJsonCheck: true,
    },
  );

  expect(exitCode).toBe(0);
  expect(stderr).not.toContain(MULTIPLE_CONFIGS_WARNING_TEXT);
});
