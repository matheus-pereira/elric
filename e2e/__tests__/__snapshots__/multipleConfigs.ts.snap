// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple configs will warn 1`] = `
● Multiple configurations found:
    * <rootDir>/e2e/multiple-configs/elric.config.js
    * <rootDir>/e2e/multiple-configs/elric.config.json
    * \`elric\` key in <rootDir>/e2e/multiple-configs/package.json

  Implicit config resolution does not allow multiple configuration files.
  Either remove unused config files or select one explicitly with \`--config\`.

  Configuration Documentation:
  https://elricjs.io/docs/configuration.html

PASS Config from js file __tests__/test.js
  ✓ dummy test
`;

exports[`multiple configs will warn 2`] = `
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;
